src/azure-cli-core/azure/cli/core/extension/_resolve.py

```diff
@@ -104,6 +104,12 @@ def resolve_from_index(extension_name, cur_version=None, index_url=None, target_
     if not candidates:
         raise NoExtensionCandidatesError(f"No extension found with name '{extension_name}'")
 
+    if not allow_preview:
+        stable_candidates = list(filter(is_stable_from_metadata, candidates))
+        if len(stable_candidates) == 0:
+            logger.warning("No stable version of '%s' to install. Preview versions allowed", extension_name)
+            allow_preview = True
+
```

Comment:
Two questions:
1. The following logic can be removed?
https://github.com/Azure/azure-cli/blob/7432856f7f16aec72abc9985558c5e7972ca2f40/src/azure-cli-core/azure/cli/core/extension/_resolve.py#L124-L130
2. If user has pass `--allow-preview false`, we still set it to True. It's kind of weird. A better solution is set default value to None, if the value is None, we can set it to True. If it's False, we keep it.

Reply:
Yes I agree the fallback logic here for those extension modules who do not have any stable versions is a little bit odd. 

Previously I'm thinking add a new config entry for this behavior so that, if needed, this behavior can be ignored. But that seems adding details on it.

However, what we want for all extension modules, is installing stable versions by default, not allow preview versions. So setting `allow-preview` to be `None` and resetting it to be `True` for `None` is not what we designed it for.